snowballDevs/kandoo
# Implemented the feature to allow users to create tasks on the board

## Description

Enhancements and Refactoring for Task Management

**New Task Creation**:
- Implemented the ability for users to create new tasks within the application.
- Sent task creation data to the backend for persistence, ensuring newly created tasks now reside in the database.
- Enhanced the `createTask` controller function to respond with the newly created task.

**Organizational Changes**:
- Relocated the "Add Task" component to the `kanban` folder in feagtures for improved project structure and organization.
- Refactoring  of the `updateTask` and `deleteTask` functions within the Workspace Slideover to ensure they update the correct state (specifically, `items` and not `selectedBoard`).


## Related Issue

-   [ ] Is this related to a specific issue? Is so please specify:

## Type of Changes

Use a check for ✓ the following type of changes:

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|   ✓  | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates


### After
Can see here the successful creation of a new task
![image](https://github.com/snowballDevs/kandoo/assets/105207353/f180fcaf-4cbc-4b9c-9d6a-7f73c22db0d5)

## Testing Steps / QA Criteria

- Run react dev tools
- Create a task in the workspace
- Check the items state in boardContext to see that state is being updated correctly
- Check the database that the task is being created there

## Checklist:
